network/transport/nats/nats_connection.go

```diff
@@ -1,73 +1,50 @@
 package nats
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/nats-io/nats.go"
 	"github.com/rs/zerolog/log"
+	"github.com/statechannels/go-nitro/network/serde"
 )
 
 type natsConnection struct {
-	nc *nats.Conn
-
-	subTopicNames     []string
-	msgChannel        chan *nats.Msg
+	nc                *nats.Conn
 	natsSubscriptions []*nats.Subscription
 }
 
-func NewNatsConnection(nc *nats.Conn, subTopicNames []string) *natsConnection {
+func NewNatsConnection(nc *nats.Conn) *natsConnection {
 	natsConnection := &natsConnection{
 		nc:                nc,
-		subTopicNames:     subTopicNames,
-		msgChannel:        make(chan *nats.Msg, 128),
-		natsSubscriptions: make([]*nats.Subscription, len(subTopicNames)),
+		natsSubscriptions: make([]*nats.Subscription, 0),
 	}
-	go natsConnection.subscribeToTopics()
 
 	return natsConnection
 }
 
-func (c *natsConnection) subscribeToTopics() {
-	for _, topic := range c.subTopicNames {
-		sub, err := c.subscribeToTopic(topic, 0)
-		if err != nil {
-			log.Error().Err(err).Msgf("failed to connect on topic: %s", topic)
-		} else {
-			c.natsSubscriptions = append(c.natsSubscriptions, sub)
-		}
-	}
-}
-
-func (c *natsConnection) subscribeToTopic(topic string, try int32) (*nats.Subscription, error) {
-	sub, err := c.nc.ChanSubscribe(topic, c.msgChannel)
-
-	if err != nil && try < 3 {
-		time.Sleep(time.Millisecond * 100)
-		return c.subscribeToTopic(topic, try+1)
-	}
-
-	return sub, nil
-}
-
-func (c *natsConnection) Send(t string, data []byte) {
-	err := c.nc.Publish(t, data)
-	if err != nil {
-		log.Error().Err(err).Msgf("failed to send message on topic: %s. msg: %s", t, string(data))
+func (c *natsConnection) Request(topic serde.RequestMethod, data []byte) ([]byte, error) {
+	msg, err := c.nc.Request(methodToTopic(topic), data, 10*time.Second)
+	if msg == nil {
+		return nil, fmt.Errorf("received nill data for request %v with error %w", topic, err)
 	}
-	log.Trace().Msgf("published message on %s.\ndata: %v", t, string(data))
+	return msg.Data, err
 }
 
-func (c *natsConnection) Recv() ([]byte, error) {
-	msg := <-c.msgChannel
+func (c *natsConnection) Subscribe(topic serde.RequestMethod, handler func([]byte) []byte) error {
```

Comment:
nit, godoc

Reply:
Good call! Done in [b2c9f63](https://github.com/statechannels/go-nitro/pull/1102/commits/b2c9f63b3d04ac5c1dd8ed306883b2b887728d69)